integration_tests/broker_test.go

```diff
@@ -153,33 +245,85 @@ var _ = Describe("Broker", func() {
-		Expect(string(body)).To(ContainSubstring("Can't update an instance"))
+		body := readResponseBody(responseRecorder.Body)
+		Expect(string(body)).To(ContainSubstring("changing plans is not currently supported"))
 	})
 
-	It("serves the last operation endpoint", func() {
-		path := fmt.Sprintf("/v2/service_instances/%s/last_operation", instanceID)
-		req := newRequest(
-			"GET",
-			path,
-			nil,
-			uriParam{key: "service_id", value: serviceID},
-			uriParam{key: "plan_id", value: planID},
-			uriParam{key: "operation", value: "some ID returned by an async operation"},
+	Context("when checking the status of the last operation", func() {
+		var (
+			path string
+			req  *http.Request
 		)
-		resp := serveRequest(responseRecorder, req)
-		Expect(responseRecorder.Code).To(Equal(http.StatusInternalServerError))
-		body := readResponseBody(resp)
-		Expect(string(body)).To(ContainSubstring("Can't check last operation"))
+
+		BeforeEach(func() {
+			path = fmt.Sprintf("/v2/service_instances/%s/last_operation", instanceID)
+			req = newRequest(
+				"GET",
+				path,
+				nil,
+				uriParam{key: "service_id", value: serviceID},
+				uriParam{key: "plan_id", value: planID},
+				uriParam{key: "operation", value: "recipe-id"},
+			)
+		})
+
+		It("returns an error when unable to get the recipe", func() {
+			fakeComposeClient.GetRecipeErr = fmt.Errorf("error: failed to get recipe by ID")
+			brokerAPI.ServeHTTP(responseRecorder, req)
+			Expect(fakeComposeClient.GetRecipeID).To(Equal("recipe-id"))
+			Expect(responseRecorder.Code).To(Equal(http.StatusInternalServerError))
+			body := readResponseBody(responseRecorder.Body)
+			Expect(string(body)).To(ContainSubstring(`{"description":"error: failed to get recipe by ID"}`))
+		})
+
+		It("returns a failed state when the Compose recipe status is not recognised", func() {
+			fakeComposeClient.GetRecipeStatus = "some-unknown-recipe-status"
+			brokerAPI.ServeHTTP(responseRecorder, req)
+			Expect(fakeComposeClient.GetRecipeID).To(Equal("recipe-id"))
+			Expect(responseRecorder.Code).To(Equal(http.StatusOK))
+			body := readResponseBody(responseRecorder.Body)
+			Expect(string(body)).To(ContainSubstring("failed"))
+		})
+
+		It("returns OK when last operation has completed", func() {
+			fakeComposeClient.GetRecipeStatus = "complete"
+			brokerAPI.ServeHTTP(responseRecorder, req)
+			Expect(fakeComposeClient.GetRecipeID).To(Equal("recipe-id"))
+			Expect(responseRecorder.Code).To(Equal(http.StatusOK))
+			body := readResponseBody(responseRecorder.Body)
+			Expect(string(body)).To(ContainSubstring("succeeded"))
+		})
+
+		// According to the CF Services API this should return 202 Accepted[1],
+		// but the brokerapi library does not currently check the state and returns 200 OK[2] regardless.
+		// [1] https://docs.cloudfoundry.org/services/api.html#updating_service_instance
+		// [2] https://github.com/pivotal-cf/brokerapi/blob/8f25c50c2794f9401a0162e260801e877ea3d002/api.go#L380-L385
```

Comment:
It appears that it's actually `200` and the brokerapi is returning the proper thing: https://docs.cloudfoundry.org/services/api.html#polling